MathLib/MathTools.h

```diff
@@ -111,6 +115,16 @@ double sqrDist(const double* p0, const double* p1)
  */
 double getAngle (const double p0[3], const double p1[3], const double p2[3]);
 
+/// sign function
+template <typename T> int sgn(T val) {
+    return (T(0) < val) - (val < T(0));
+}
```

Comment:
The sign function is already implemented in [boost](http://www.boost.org/doc/libs/1_50_0/libs/math/doc/sf_and_dist/html/math_toolkit/utils/sign_functions.html): `#include <boost/math/special_functions/sign.hpp>`
I'd prefer an existing function